climate_police/Homework7/pollution.py

```diff
@@ -0,0 +1,103 @@
+"""pollution.py"""
+
+import wget
+import os
+import zipfile
+
+import urllib3
+import certifi
+import sys
+import glob
+
+import numpy as np
+import pandas as pd
+
+import matplotlib 
+import matplotlib.pyplot as plt
+import plotly.offline as py
+import plotly.graph_objs as go
+import plotly.tools as tls
+import seaborn as sns
+
+import plotly
+plotly.offline.init_notebook_mode() 
+
+
+def download_if_needed(URL, filename):
+    """
+    Download from URL to filename unless filename already exists
+    """
+
+    if os.path.exists(filename):
+        pass
+    else:
+        print('downloading', filename,'This may take several seconds... Please be patient :)')
+
+        # needed for https
+        http = urllib3.PoolManager(
+            cert_reqs='CERT_REQUIRED', # Force certificate check.
+            ca_certs=certifi.where(),  # Path to the Certifi bundle.
+            retries=3
+        )
+
+        try:
+            response = http.request('GET',URL)
+            with open(filename,'wb') as f:
+                f.write(response.data)
+            f.close()
+            response.release_conn()
+
+        except:
+            e = sys.exc_info()[0]
+            print('Error! Data not downloaded.',e)
+
+def get_pollutants_data():
+    """
+    Download temperature data, unless already downloaded
+    """
+    download_if_needed("https://kaggle2.blob.core.windows.net/datasets/312/676/pollution_us_2000_2016.csv.zip",'pollutants.zip')
```

Comment:
Downloading requires login, so maybe we can put the data onto a public server, like dropbox's public links.